FC-Chess/FC-Chess.cpp

```diff
@@ -107,28 +105,271 @@ void FCC::Chess::checkMovesHorizontal()
 {
     for (uint8_t i = selX; i <= 7; i++)
     {
-        if (checkIsEnemyOnField(&i, &selY)) break;
+        if (checkIsFriendlyOnField(&i, &selY)) break;
         posPossible[i][selY] = true;
+        if (checkIsEnemyOnField(&i, &selY)) break;
     }
     for (uint8_t i = selX; i >= 0; i--)
     {
-        if (checkIsEnemyOnField(&i, &selY)) break;
+        if (checkIsFriendlyOnField(&i, &selY)) break;
         posPossible[i][selY] = true;
+        if (checkIsEnemyOnField(&i, &selY)) break;
     }
 }
 
 void FCC::Chess::checkMovesVertical()
 {
     for (uint8_t i = selY; i <= 7; i++)
     {
-        if (checkIsEnemyOnField(&selX, &i)) break;
+        if (checkIsFriendlyOnField(&selX, &i)) break;
         posPossible[selX][i] = true;
+        if (checkIsEnemyOnField(&selX, &i)) break;
     }
     for (uint8_t i = selY; i >= 0; i--)
     {
-        if (checkIsEnemyOnField(&selX, &i)) break;
+        if (checkIsFriendlyOnField(&selX, &i)) break;
         posPossible[selX][i] = true;
+        if (checkIsEnemyOnField(&selX, &i)) break;
+    }
+}
+
+void FCC::Chess::checkMovesDiagonal()
+{
+    // creating variables for x and y
+    uint8_t i = selX, j = selY;
+
+    // checking possible moves
+    for (; i <= 7 && j <= 7;)
+    {
+        if (checkIsFriendlyOnField(&i, &j)) break;
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i++;
+        j++;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    for (; i <= 7 && j >= 0;)
+    {
+        if (checkIsFriendlyOnField(&i, &j)) break;
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i++;
+        j--;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    for (; i >= 0 && j <= 7;)
+    {
+        if (checkIsFriendlyOnField(&i, &j)) break;
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i--;
+        j++;
+    }
+    
+    i = selX, j = selY;     // resetting variables
+    for (; i >= 0 && j >= 0;)
```

Comment:
same as before with:
```
if (checkIsFriendlyOnField(i - selX, j - selY )) break;
```

Reply:
applied that with the upcoming removing pointers push